sources/pmwebapi_test.go

```diff
@@ -55,49 +55,41 @@ func TestUnmarshal(t *testing.T) {
 	}
 }
 
-func TestLabelTypes(t *testing.T) {
-	units := ""
-	pcpType := ""
-	name := "kal"
-	response := typeLabel(units, pcpType, name)
-	expected := "kal"
-	if response != expected {
-		t.Fatalf("Failed when units and type strings were empty. Expected: %s, Got: %s", expected, response)
-	}
-
-	units = ""
-	pcpType = ""
-	name = ""
-	response = typeLabel(units, pcpType, name)
-	expected = ""
-	if response != expected {
-		t.Fatalf("Failed when all strings were empty. Expected: %s, Got: %s", expected, response)
-	}
+func TestWeirdMetrics(t *testing.T) {
+	name := make(map[string]string)
 
-	units = "kal"
-	pcpType = "CoUnTeR"
-	name = "kal"
-	response = typeLabel(units, pcpType, name)
-	expected = "kal_kal_total"
-	if response != expected {
-		t.Fatalf("Failed when type was case sensitive. Expected: %s, Got: %s", expected, response)
+	name["count / sec"] = "count_per_second"
+	name["Kbyte"] = "kilobytes"
+	name["nanosec"] = "nanoseconds"
+	name["millisec"] = "milliseconds"
+	name["failcnt"] = "failcount"
+	name["byte"] = "bytes"
+	name["Mbyte"] = "megabytes"
+	name["sec"] = "seconds"
+	name["bytes_byte"] = "bytes"
+	name["kilobytes"] = "kilobytes"
+	name["seconds"] = "seconds"
+	name["nanoseconds"] = "nanoseconds"
+	name["milliseconds"] = "milliseconds"
+	name["megabytes"] = "megabytes"
+	name["bytes"] = "bytes"
+	for k := range name {
+		if fixNaming(k) != name[k] {
+			t.Fatalf("Failed when name was %s. Expected: %s, Got: %s", k, name[k], fixNaming(k))
```

Comment:
Remember to set the result of `fixNaming(k)` to a variable before the `if` statement, and compare that instead of making the call in this function again. This will prevent any weirdness with a return or errors.